iothub_client/inc/iothub_device_client.h

```diff
@@ -349,6 +348,88 @@ extern "C"
     */
     MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync, IOTHUB_DEVICE_CLIENT_HANDLE, iotHubClientHandle, const char*, destinationFileName, IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_CALLBACK_EX, getDataCallbackEx, void*, context);
 
+     /**
+    * @brief    This API creates a context for a new blob upload to Azure Storage.
+    * @remark   It is part of a set of functions for more granular control over Azure IoT-based blob uploads.
+    *           This function is expected to be used along with:
+    *           `IoTHubDeviceClient_CreateUploadContext`
+    *           `IoTHubDeviceClient_UploadBlockToBlob`
+    *           `IoTHubDeviceClient_DestroyUploadContext`
+    *           For the standard less-granular uploads to blob please use either
+    *           `IoTHubDeviceClient_UploadToBlobAsync` or `IoTHubDeviceClient_UploadMultipleBlocksToBlobAsync`.
+    *
+    * @param    iotHubClientHandle      The handle created by a call to the create function.
+    * @param    destinationFileName     name of the file.
+    *
+    * @warning  This is a synchronous/blocking function.
+    *
+    * @return   A `IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE` on success or NULL if the function fails.
```

Comment:
LL functionality mixed with non-LL is confusing. Please create another wrapping type for non-LL (a type-def is fine as it would allow us to extend in the future since we don't provide ABI compat).

Reply:
Our idea: 

iothub_device_client.h
typedef IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE IOTHUB_CLIENT_AZURE_BLOB_CLIENT_HANDLE;

iothub_device_client_ll.h
typedef IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE IOTHUB_CLIENT_LL_AZURE_BLOB_CLIENT_HANDLE;

Reply:
Done.